packages/database/src/migrations/20210317035935-AddConfigForCategoryHeadingLaosEoc-modifies-data.js

```diff
@@ -0,0 +1,163 @@
+'use strict';
+
+import { updateValues } from '../utilities/migration';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+async function getDashboardReportById(db, id) {
+  const { rows: dashboardReports } = await db.runSql(`
+      SELECT * FROM "dashboardReport"
+      WHERE id = '${id}';
+  `);
+  return dashboardReports[0] || null;
+}
+
+const categoryPresentationOptions = {
+  conditions: {
+    red: {
+      color: '#b71c1c',
+      legendLabel: 'Stock out',
+    },
+    green: {
+      color: '#33691e',
+      label: '',
+      legendLabel: 'In stock',
+    },
+    orange: {
+      color: 'orange',
+      label: '',
+      legendLabel: 'At least 1 item out of stock',
+    },
+  },
+  showRawValue: true,
+  showNestedRows: true,
+};
+
+const previousPresentationOptions = {
+  type: 'condition',
+  conditions: [
+    {
+      key: 'red',
+      color: '#b71c1c',
+      label: '',
+      condition: 0,
+      description: 'Stock number: ',
+      legendLabel: 'Stock out',
+    },
+    {
+      key: 'green',
+      color: '#33691e',
+      label: '',
+      condition: {
+        '>': 0,
+      },
+      description: 'Stock number: ',
+      legendLabel: 'In stock',
+    },
+  ],
+  showRawValue: true,
+};
+
+const newPresentationOptions = {
+  type: 'condition',
+  conditions: [
+    {
+      key: 'red',
+      color: '#b71c1c',
+      label: 'Stock number: 0',
+      condition: {
+        '=': '',
+      },
+      legendLabel: 'Stock out',
+    },
+    {
+      key: 'green',
+      color: '#33691e',
+      label: '',
+      condition: {
+        '>': 0,
+      },
+      description: 'Stock number: ',
+      legendLabel: 'In stock',
+    },
+    {
+      key: 'orange',
+      color: 'orange',
+      label: '',
+      legendLabel: 'At least 1 item out of stock',
+    },
+  ],
+  showRawValue: true,
+};
+
+const categoryAggregator = {
+  type: '$condition',
+  conditions: [
+    {
+      key: 'red',
+      condition: {
+        '=': null,
```

Comment:
What if one value is `0` and the others `null`, or all values are `0`? Shouldn't we use the `red` color in that case too?

Reply:
Oh I see what's happening... `0` values will be converted to `null` in [buildCategoryData](https://github.com/beyondessential/tupaia/blob/50ed7d3d7bc65a2a21ea3b359b6e4aa4c042ee05/packages/web-config-server/src/apiV1/dataBuilders/generic/table/tableOfDataValues/helpers/buildCategoryData.js#L52).

In general, this is a bad practice: 0 means that we **know** that the data equals to zero, while `null`/`undefined` means we do **not know** the data. Both of them are treated the same here, but this is a happy coincidence - more often than not this is not the case. Moreover, future devs may get confused, as I did while looking at the config.

It seems that we have support for `in` operations in `checkValueSatisfiesCondition`, so something like the following may work out of the box:

```js
{
  condition: { 'in': [0, null] }
}
```